credentials/auth.go

```diff
@@ -32,21 +34,38 @@ type netlifyConfig struct {
 	Users       map[string]netlifyUserInfo `json:"users,omitempty"`
 }
 
-var validAuthPaths = [][]string{
-	{".netlify", "config.json"},
-	{".config", "netlify"},
-	{".netlify", "config"},
-	{".config", "netlify.json"},
+func getValidAuthPaths() ([][]string, error) {
+	home, err := homedir.Dir()
+	if err != nil {
+		return nil, err
+	}
+
+	configPaths := []string{xdg.ConfigHome, "netlify", "config.json"}
+	if goruntime.GOOS == "windows" {
+		// on windows the go xdg lib behaves a bit different than the Node.js one
+		// https://github.com/sindresorhus/env-paths/blob/de22adb240117c7aacf0365187472907a9e06872/index.js#L28
+		// https://github.com/adrg/xdg/blob/af0f1bbdcb2e9415a67e57c4125afc9daeb3ca17/paths_windows.go#L40
+		configPaths = []string{xdg.DataDirs[0], "netlify", "Config", "config.json"}
+	}
+
+	validAuthPaths := [][]string{
+		configPaths,
+		{home, ".netlify", "config.json"},
+		{home, ".config", "netlify"},
+		{home, ".netlify", "config"},
+		{home, ".config", "netlify.json"},
+	}
+
+	return validAuthPaths, nil
 }
 
 func saveAccessToken(token string) error {
-	home, err := homedir.Dir()
+	validAuthPaths, err := getValidAuthPaths()
 	if err != nil {
 		return err
 	}
 
-	args := append([]string{home}, validAuthPaths[0]...)
-	f, err := os.OpenFile(filepath.Join(args...), os.O_CREATE|os.O_RDWR, 0644)
+	f, err := os.OpenFile(filepath.Join(validAuthPaths[0]...), os.O_CREATE|os.O_RDWR, 0644)
```

Comment:
Question - should we make sure to created the nested directory path if the file doesn't exist?

Reply:
another #lazyweb but what `xdg.ConfigHome` looks like usually? (or `APPDATA`)
I feel like at least the `netlify` (and `Config` for windows) directory needs to be created regardless?
how are we doing in CLI side?

Reply:
Commented on the structure in https://github.com/netlify/netlify-credential-helper/pull/51#discussion_r573416440.

The CLI uses [configstore](https://github.com/yeoman/configstore) to access the file so it handles directory creation https://github.com/yeoman/configstore/blob/ab81102b4917fda25d2e2ec616a00ea9e9227a36/index.js#L59

Reply:
Added another commit for this https://github.com/netlify/netlify-credential-helper/pull/51/commits/1497e367691b37489f8d33c3df11491cf0f2dec2

A follow up question - should we change the permissions for the file/directories to `0600` or `0640` to prevent others reading the file?

Reply:
looks like directories are 700 and file is 600 with CLI (aka configstore)?

https://github.com/yeoman/configstore/blob/ab81102b4917fda25d2e2ec616a00ea9e9227a36/index.js#L13-L14

I'd vote for being as consistent as possible with CLI here.

Reply:
Done in https://github.com/netlify/netlify-credential-helper/pull/51/commits/2978889a69d51ee6012eafb93ee924efdafa09d0